packages/utils/src/baggage.ts

```diff
@@ -120,3 +140,26 @@ export function mergeAndSerializeBaggage(incomingBaggage?: Baggage, headerBaggag
   );
   return serializeBaggage(finalBaggage);
 }
+
+/**
+ * Helper function that takes a raw baggage string (if available) and the processed sentry-trace header
+ * data (if available), parses the baggage string and creates a Baggage object
+ * If there is no baggage string, it will create an empty Baggage object.
+ * In a second step, this functions determines when the created Baggage object should be set immutable
+ * to prevent mutation of the Sentry data.
+ *
+ * Extracted this logic to a function because it's duplicated in a lot of places.
+ *
+ * @param rawBaggageString
+ * @param traceparentData
+ */
+export function parseBaggageSetMutability(
+  rawBaggageString: string | false | undefined | null,
+  traceparentData: TraceparentData | string | false | undefined | null,
+): Baggage {
+  const baggage = parseBaggageString(rawBaggageString || '');
+  if (!isSentryBaggageEmpty(baggage) || (traceparentData && isSentryBaggageEmpty(baggage))) {
```

Comment:
Just because I haven't quite groked yet why we set the baggage to immutable here, if there is traceparentData: Is traceparent data connected to baggage in some way? Could you quickly explain here or maybe even add a comment?

Reply:
Sure, so the thing is that `traceParentData` is the processed/parsed `sentry-trace` header. `sentry-trace` and `baggage` go hand in hand which is why we have to consider both of them if we make the mutability descision (see PR description). I used `traceParentData` instead of the raw header string because I think it saves us a few bytes of bundle. 

Your comment made me realize though that I was thinking that I should change the `traceparentData` param name. Will do that now.